assembly/pallets/balances/assembly/balances-module.ts

```diff
@@ -0,0 +1,99 @@
+        } else {
+            return AccountData.getDefault();
+        }
+    }
+
+    /**
+     * @description Sets the balances of a given AccountId
+     * Alters the Free balance and Reserved balances in Storage.
+     */
+    static setBalance(accountId: AccountIdType, freeBalance: Balance, reservedBalance: Balance): AccountData {
+        const currentAccountData = this.getAccountData(accountId);
+
+        // TODO Any meaningful checks
+
+        currentAccountData.setFree(freeBalance);
+        currentAccountData.setReserved(reservedBalance);
+        Storage.set(accountId.getAddress(), currentAccountData.toU8a());
+
+        return currentAccountData;
+    }
+
+    /**
+     * @description Transfer the given amount from sender to receiver
+     * Note: this is just draft implementation, without necessary checks
+     * @param sender sender account
+     * @param receiver receiver account
+     * @param amount amount of the transfer
+     */
+    static transfer(sender: AccountIdType, receiver: AccountIdType, amount: Balance): void {
+        const senderAccData = this.getAccountData(sender);
+        const receiverAccData = this.getAccountData(receiver);
+        const senderNewBalance: Balance = instantiate<Balance>(u128.sub(senderAccData.getFree().value, amount.value));
+        const receiverNewBalance: Balance = instantiate<Balance>(u128.sub(receiverAccData.getFree().value, amount.value));
+        this.setBalance(sender, senderNewBalance, senderAccData.getReserved());
+        this.setBalance(receiver, receiverNewBalance, receiverAccData.getReserved());
+        System.incAccountNonce(sender);
+        Log.info("Done transfering: " + amount.toString());
+    }
+
+    /**
+     * @description Apply extrinsic for the module
+     * @param extrinsic SignedTransaction instance
+     */
+    static applyExtrinsic(extrinsic: ISignedTransaction): u8[]{
+        const sender: AccountIdType = BytesReader.decodeInto<AccountIdType>(extrinsic.getFrom().toU8a());
+        const receiver: AccountIdType = BytesReader.decodeInto<AccountIdType>(extrinsic.getTo().toU8a());
+        const validated = this.validateTransaction(extrinsic);
+        if(!validated.valid){
+            Log.error(validated.message);
+            return validated.error;
+        }
+        this.transfer(sender, receiver, BytesReader.decodeInto<Balance>(extrinsic.getAmount().toU8a()));
+        return ResponseCodes.SUCCESS;
+    }
+
+    /**
+     * @description Validate transaction before applying it 
+     * @param extrinsic SignedTransaction instance
+     */
+    static validateTransaction(extrinsic: ISignedTransaction): TransactionValidity{
+        const from: AccountIdType = BytesReader.decodeInto<AccountIdType>(extrinsic.getFrom().toU8a());
+        const fromBalance = Balances.getAccountData(from);
+        const balance: Balance = fromBalance.getFree();
+        if(balance.value < u128.fromU64((<UInt64>extrinsic.getAmount()).value)){
```

Comment:
Extrinsics (amount) should be the same type as free/reserved balances.
We have the same problem here. We are assuming that the balance type is u128. Maybe we need to move more models into `subsembly` in order to use the user-defined `Balance` type